File: python sandbox/factors.go
package main

import (
	"fmt"
	"math"
	"sort"
	"time"
)

// i wrote this in Golang, just for the speed in execution, prototyping was done in python itself.

func findFactors(number int) []int {
	factors := []int{}
	sqrtNum := int(math.Sqrt(float64(number)))

	for i := 1; i <= sqrtNum; i++ {
		if number%i == 0 {
			factors = append(factors, i)
			otherFactor := number / i
			if otherFactor != i {
				factors = append(factors, otherFactor)
			}
		}
	}
	return factors
}

func main() {
	num := 999999999999999999
	startTime := time.Now()

	result := findFactors(num)
	sort.Ints(result)

	elapsedTime := time.Since(startTime)
	fmt.Printf("Factors of %d: %v\n", num, result)
	fmt.Printf("Elapsed time: %v\n", elapsedTime)
}

File: python sandbox/factors.py
import time
import math


def find_factors(number):
    factors = []
    square_root = math.ceil(math.sqrt(number))

    for i in range(1, (square_root) + 1):
        if number % i == 0:
            factors.append(i)
    factors.append(number)
    return factors


num = 999999999999999999
start_time = time.time()
result = find_factors(num)
end_time = time.time()
print(f"Factors of {num}: {result}")
elapsed_time_s = end_time - start_time
print(f"Elapsed time: {elapsed_time_s:.2f} seconds")
